flux/shape.py

```diff
@@ -168,7 +168,7 @@ def check_vis_1_to_N(self, i, J, eps=None):
             rayhit.prim_id == J
         )
 
-    def get_direct_irradiance(self, F0, Dsun, eps=None):
+    def get_direct_irradiance(self, F0, Dsun, basemesh=None, eps=None):
```

Comment:
Is this what Erwan was asking about (having another mesh to do check visibility only against)? If so, thanks for adding it! And you could you add a line to the docstring for this function explaining it?

Reply:
Or is this just a way to call `get_direct_irradiance` on a different 3D mesh, other than one held by the current instance of `TrimeshShapeModel`?

Reply:
`having another mesh to do check visibility only against`
precisely so, it's a second, larger mesh (as it is implemented now, the full input map) which is just used to check visibility. Sure I can expand the docstring, I forgot about it.